integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/l0promotion/InternalSdxSshAndCmAccessTest.java

```diff
@@ -0,0 +1,144 @@
+                .withClouderaManager(clouderaManager)
+                .given(RecipeTestDto.class).withName(recipeName).withContent(generateRecipeContent())
+                .withRecipeType(POST_CLOUDERA_MANAGER_START)
+                .when(recipeTestClient.createV4())
+                .given(masterInstanceGroup, InstanceGroupTestDto.class).withHostGroup(MASTER).withNodeCount(1).withRecipes(recipeName)
+                .given(idbrokerInstanceGroup, InstanceGroupTestDto.class).withHostGroup(IDBROKER).withNodeCount(1).withRecipes(recipeName)
+                .given(stack, StackTestDto.class).withCluster(cluster).withInstanceGroups(masterInstanceGroup, idbrokerInstanceGroup)
+                .given(sdxInternal, SdxInternalTestDto.class)
+                .withCloudStorage(getCloudStorageRequest(testContext))
+                .withStackRequest(key(cluster), key(stack))
+                .when(sdxTestClient.createInternal(), key(sdxInternal))
+                .await(SdxClusterStatusResponse.RUNNING)
+                .awaitForInstance(getSdxInstancesHealthyState())
+                .then((tc, testDto, client) -> sshJUtil.checkFilesOnHostByNameAndPath(testDto, client, List.of(MASTER.getName(), IDBROKER.getName()),
+                            filePath, fileName, 1, null, null))
+                .then((tc, testDto, client) -> clouderaManagerUtil.checkClouderaManagerKnoxIDBrokerRoleConfigGroups(tc, testDto, client))
+                .validate();
+
+        testContext
+                .given(freeIpa, FreeIpaTestDto.class)
+                .when(freeIpaTestClient.describe(), key(freeIpa))
+                .given(FreeIpaUserSyncTestDto.class)
+                .when(freeIpaTestClient.getLastSyncOperationStatus())
+                .await(OperationState.COMPLETED)
+                .given(UmsTestDto.class).assignTarget(EnvironmentTestDto.class.getSimpleName())
+                .when(umsTestClient.setWorkloadPassword(newWorkloadPassword))
+                .given(FreeIpaUserSyncTestDto.class)
+                .when(freeIpaTestClient.syncAll())
+                .await(OperationState.COMPLETED)
+                .given(sdxInternal, SdxInternalTestDto.class)
+                .then((tc, testDto, client) -> sshJUtil.checkFilesOnHostByNameAndPath(testDto, client, List.of(MASTER.getName()),
+                            filePath, fileName, 1, workloadUsername, newWorkloadPassword))
+                .then((tc, testDto, client) -> clouderaManagerUtil.checkClouderaManagerUserDetails(tc, testDto, client, workloadUsername, newWorkloadPassword))
+                .validate();
+    }
+
+    private String generateRecipeContent() throws IOException {
+        String recipeContentFromFile = ResourceUtil.readResourceAsString(applicationContext, getRecipePath());
+        return Base64.encodeBase64String(recipeContentFromFile.getBytes());
+    }
+
+    private String getWorkloadUsername(TestContext testContext) {
```

Comment:
do we necessarily need this method? I think we should configure the sso user for this test from parameter, but checking for it is not necessary, test will fail later anyway because of missing user

Reply:
User has been configured at https://github.com/hortonworks/cloudbreak/blob/dae4debd77cf45027ba34b41761958ec5a91ea11/integration-test/src/main/resources/application.yml#L43-L47
However the Workload Username is not present here and I do not think we should add this as a parameter, because of we can get this from Manowar.

Reply:
and what if we do not have such sso user `csso_cloudbreak-qe` in one of our environment? Anyway I would avoid hardcoding an user in the test code, if the email or something changes, we need a new version

Reply:
in my opinion the user is a prerequisite for test, if the user is not exists, the test would fail anyway, so  that's why I think we could spare a GRPC call if we would not check for user details here, also test code would be smaller

Reply:
`csso_cloudbreak-qe` just a default/sample value here what we are overwriting with the real acting workload username based on 
```workloadUsername.set(client.getDefaultClient().getUserDetails(userCrn, userCrn, Optional.of("")).getWorkloadUsername());```
So this is not a hardcoded/static value here. This is the result of the getter from UMS Test Client.

Reply:
i am sorry, you are right, cool, then this is ok for me